redisinsight/ui/src/pages/rdi/connection-form/ConnectionForm.spec.tsx

```diff
@@ -0,0 +1,131 @@
+    expect(render(<ConnectionForm {...mockedProps} />)).toBeTruthy()
+  })
+
+  it('should disable submit button when form is invalid', async () => {
+    render(<ConnectionForm {...mockedProps} />)
+
+    // click submit to trigger form validation
+    fireEvent.click(screen.getByRole('button', { name: 'Add Instance' }))
+
+    await waitFor(() => expect(screen.getByRole('button', { name: 'Add Instance' })).toBeDisabled(), {
+      timeout: 1000
+    })
+  })
+
+  it('should disable test connection button when form is invalid', async () => {
+    render(<ConnectionForm {...mockedProps} />)
+
+    // click submit to trigger form validation
+    fireEvent.click(screen.getByRole('button', { name: 'Add Instance' }))
+
+    await waitFor(() => expect(screen.getByRole('button', { name: 'Test Connection' })).toBeDisabled(), {
+      timeout: 1000
+    })
+  })
+
+  it('should not disable submit button when form is valid', async () => {
+    render(<ConnectionForm {...mockedProps} />)
+
+    // click submit to trigger form validation
+    fireEvent.click(screen.getByRole('button', { name: 'Add Instance' }))
+
+    fireEvent.change(screen.getByTestId('connection-form-url-input'), { target: { value: 'url' } })
+    fireEvent.change(screen.getByTestId('connection-form-username-input'), { target: { value: 'username' } })
+    fireEvent.change(screen.getByTestId('connection-form-password-input'), { target: { value: 'password' } })
+
+    await waitFor(() => expect(screen.getByRole('button', { name: 'Add Instance' })).not.toBeDisabled(), {
+      timeout: 1000
+    })
+  })
+
+  it('should not disable submit button when form is provided editInstance', async () => {
+    render(<ConnectionForm {...mockedProps} editInstance={mockedEditInstance} />)
+
+    // click submit to trigger form validation
+    fireEvent.click(screen.getByRole('button', { name: 'Add Instance' }))
+
+    await waitFor(() => expect(screen.getByRole('button', { name: 'Add Instance' })).not.toBeDisabled(), {
+      timeout: 1000
+    })
+  })
+
+  it('should disable URL input when form is provided editInstance', () => {
+    render(<ConnectionForm {...mockedProps} editInstance={mockedEditInstance} />)
+
+    expect(screen.getByTestId('connection-form-url-input')).toBeDisabled()
+  })
+
+  it('should show url tooltip when url input is not disabled', async () => {
+    render(<ConnectionForm {...mockedProps} />)
+
+    // click submit to trigger form validation
+    fireEvent.click(screen.getByRole('button', { name: 'Add Instance' }))
+
+    fireEvent.mouseOver(screen.getByTestId('connection-form-url-icon'))
+
+    const tooltip = await screen.findByTestId('connection-form-url-tooltip')
+
+    expect(tooltip).toBeInTheDocument()
+  })
+
+  it('should show validation tooltip when submit button is disabled', async () => {
+    render(<ConnectionForm {...mockedProps} />)
+
+    // click submit to trigger form validation
+    fireEvent.click(screen.getByRole('button', { name: 'Add Instance' }))
```

Comment:
Will be better to add data-testid to the buttons and use getByTestId. And try add more data-test id properties, it will help QA to write e2e tests as well